snapshot/restore/restore.go

```diff
@@ -75,8 +81,9 @@ type Options struct {
 	Cancel           chan struct{} // channel that can be externally closed to signal cancelation
 }
 
-// Entry walks a snapshot root with given root entry and restores it to the provided output.
-func Entry(ctx context.Context, rep repo.Repository, output Output, rootEntry fs.Entry, options Options) (Stats, error) {
+// Entries restores one or more EntryPair objects using the provided output
+// by walking the EntryPair.root to the EntryPair.dest.
+func Entries(ctx context.Context, rep repo.Repository, output Output, roots []EntryPair, options Options) (Stats, error) {
```

Comment:
Do we need do do that? I think this complicates the code by a lot, and it's not obvious to me what the gain is.

I think you're trying to reuse the single Output for multiple restores - would it be simpler to run all this in a sequence and have one `Output` for each restore? In other words - encapsulate all the change in the CLI layer only.

Reply:
I thought of doing that and went down this rabbit hole of a more complicated approach so that the shallow expansions could all get queued up for parallel execution. The way that you suggest would definitely be simpler. I can do that.